openrouteservice/src/main/java/org/heigit/ors/routing/RoutingProfileManager.java

```diff
@@ -325,7 +325,8 @@ public RouteResult matchTrack(MapMatchingRequest req) throws Exception {
             throw new InternalServerException(RoutingErrorCodes.INVALID_PARAMETER_VALUE, "Alternative routes algorithm does not support more than two way points.");
         }
 
-        ExtraInfoProcessor extraInfoProcessor = null;
+        int numberOfExpectedExtraInfoProcessors = req.getSearchParameters().getAlternativeRoutesCount() < 0 ? 1 : req.getSearchParameters().getAlternativeRoutesCount();
+        ExtraInfoProcessor[] extraInfoProcessor = new ExtraInfoProcessor[numberOfExpectedExtraInfoProcessors];
```

Comment:
Would `extraInfoProcessors` be a better name for this variable?

Reply:
yep.